### ISSUE TITLE
Incorrect size in header when using custom chunks

### ISSUE THREAD

**Issue by rkat** [dancasimiro/WAV.jl#50]
In `wavwrite`, the length of the data chunk is used for both the header and the data chunk size.
```julia
data_length::UInt32 = size(samples, 1) * block_align
# ...
WAV.write_standard_header(io, data_length) # < here
# ...
WAV.write(io, b"data")
WAV.write_le(io, data_length)
WAV.write_data(io, fmt, samples)
```
This results in an incorrect size in the header in the case where custom chunks have been supplied. 
The fix looks something like this:
```julia
data_length::UInt32 = size(samples, 1) * block_align
# (4 byte id + 4 byte length + data length) for each custom chunk
extra_length = 8 * length(chunks) + sum([length(c) for (_, c) in chunks]) 
# ...
WAV.write_standard_header(io, data_length + extra_length) # < add extra length here
# ...
WAV.write(io, b"data")
WAV.write_le(io, data_length) # < this bit stays the same
WAV.write_data(io, fmt, samples)
```
Thanks for your work on this package, it's fantastic 👍

**Comment by dancasimiro** [comment]
Hi @rkat,

I'm sorry about this bug. It looks like you put together a good fix. Do you have time to submit a pull request? I think that you are only missing a unit test.

If not, I will base a fix on the code that you supplied the next chance that I have.

Thanks for the bug report!

**Comment by rkat** [comment]
No worries, I'll add a unit test and make a pull request :)

**Comment by dancasimiro** [comment]
Fixed by pull request #51, commit f0ed3cdd48591b6fd004c8932f8b1887607340d1

**Comment by dancasimiro** [comment]
Fixed in v0.9.0

**Comment by rkat** [comment]
Awesome, thanks Dan!
